### PR TITLE
add show confirmation on screen async task

### PR DESCRIPTION
**Pull request by Fredaly**
Add a show confirmation on screen Async Task for splitscreen games

### CODE REVIEW

File: Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+UAsyncAction_ShowConfirmationOnScreen * UAsyncAction_ShowConfirmationOnScreen::ShowConfirmationOnScreenYesNo( UObject * world_context_object, FText title, FText message, TSubclassOf< UCommonGameDialog > dialogue_widget, int z_order )
+{
+    return CreateAction( world_context_object, UCommonGameDialogDescriptor::CreateConfirmationYesNo( title, message ), dialogue_widget, z_order );
+}
+
+void UAsyncAction_ShowConfirmationOnScreen::Activate()
+{
+    if ( WorldContextObject && !TargetLocalPlayer )
```
Comment (Frickerson) [inline Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp]: ...!= nullptr && ...== nullptr

File: Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        else if ( const auto * world = WorldContextObject->GetWorld() )
+        {
+            if ( const auto * game_instance = world->GetGameInstance< UGameInstance >() )
+            {
+                TargetLocalPlayer = game_instance->GetPrimaryPlayerController( false )->GetLocalPlayer();
+            }
+        }
+    }
+
+    if ( TargetLocalPlayer )
```
Comment (Frickerson) [inline Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp]: != nullptr

File: Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        if ( TargetLocalPlayer->GetSubsystem< UCommonMessagingSubsystem >() )
```
Comment (Frickerson) [inline Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp]: != nullptr

File: Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            FCommonMessagingResultDelegate result_callback = FCommonMessagingResultDelegate::CreateUObject( this, &UAsyncAction_ShowConfirmationOnScreen::HandleConfirmationResult );
```
Comment (Frickerson) [inline Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp]: auto

**Review by Frickerson — CHANGES_REQUESTED** [review]

File: Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp (hunk 1 above)
Comment (Fredaly) [reply Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp]: code moved

File: Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp (hunk 2 above)
Comment (Fredaly) [reply Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp]: code moved

File: Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp (hunk 3 above)
Comment (Fredaly) [reply Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp]: code moved

File: Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp (hunk 4 above)
Comment (Fredaly) [reply Source/Private/Actions/AsyncAction_ShowConfirmationOnScreen.cpp]: code moved

**Review by Frickerson — APPROVED** [review]

**Review by TheEmidee — APPROVED** [review]

**Pull request merged**